airflow/ui/src/components/DataTable/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,188 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  FormControl,
+  FormLabel,
+  Input,
+  VStack,
+  ModalCloseButton,
+  Button,
+  ModalFooter,
+  Box,
+  Text,
+  Spacer,
+  HStack,
+  Collapse,
+} from "@chakra-ui/react";
+import { autocompletion } from "@codemirror/autocomplete";
+import { json } from "@codemirror/lang-json";
+import { oneDark } from "@codemirror/theme-one-dark";
+import CodeMirror, { lineNumbers } from "@uiw/react-codemirror";
+import React, { useState, useEffect } from "react";
+
+type DagParams = {
+  configJson: string;
+  dagId: string;
+  logicalDate: string;
+  runId?: string;
+};
+
+type TriggerDAGFormProps = {
+  dagParams: DagParams;
+  onClose: () => void;
+  onTrigger: () => void;
+  setDagParams: React.Dispatch<React.SetStateAction<DagParams>>;
+};
+
+const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
+  dagParams,
+  onTrigger,
+  setDagParams,
+}) => {
+  const [showDetails, setShowDetails] = useState(false);
+
+  useEffect(() => {
+    try {
+      const prettyJson = JSON.stringify(
+        JSON.parse(dagParams.configJson),
```

Comment:
I tried with example_params_ui_tutorial but the pre-filled dict was `{}` - seems the defaults did not come down as default.

Reply:
As discussed, this will be taken  upon API integartion.